sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -1423,7 +1429,7 @@ public TextAnalyticsClient(Uri endpoint, AzureKeyCredential credential, TextAnal
         /// <exception cref="RequestFailedException">Service returned a non-success
         /// status code.</exception>
         public virtual async Task<AnalyzeHealthcareEntitiesOperation> StartAnalyzeHealthcareEntitiesAsync(IEnumerable<string> documents, string language = default, AnalyzeHealthcareEntitiesOptions options = default, CancellationToken cancellationToken = default) =>
-            await _serviceClient.StartAnalyzeHealthcareEntitiesAsync(documents, language, options, cancellationToken).ConfigureAwait(false);
+            await _serviceClient.StartAnalyzeHealthcareEntitiesAsync(documents, language, options?.CheckSupported(ServiceVersion), cancellationToken).ConfigureAwait(false);
```

Comment:
This feels a little strange to me that the service method is what was newly added, but we are validating the model properties rather than the version the service was added.  I would expect us to establish some sort of precedence hierarchy if we decide to do validation, and make sure we're doing the least-expensive check first.

Reply:
Actually, the method is as well and the check would be redundant in a few other cases, but not this one: `AnalyzeHealthcareEntitiesOptions` also works for the legacy service. But your point is spot-on for single- and multi-label classify, as well as custom entity recognition. If I hadn't already, I'll be sure to remove the checks since the methods themselves are validated. If they are needed in the future, they should be added then.

Reply:
Turns out I already don't call `CheckSupported()` on these options classes.